### ISSUE TITLE
Documentation update suggestion for External Tools winscp

### ISSUE THREAD

**Issue by RolfWojtech** [mRemoteNG/mRemoteNG#2308]
This isn't really a bug, more of a documentation oversight:

I have setup winscp as an external tool as outlined in https://mremoteng.readthedocs.io/en/latest/external_tools_cheat_sheet.html#id1

I tried to use a winscp to a machine with a non-standard SSH port and a password that happens to include an exclamation mark.

First part of the documentation error is that the suggested string omits the port:
`scp://%Username%:%Password%@%Hostname%/`
correct use of the port would be:
`scp://%Username%:%Password%@%Hostname%:%Port%/`

This change makes winscp attempt to connect to the correct port but I was prompted for the password despite it being stored and normally working directly.
So with some luck and google I found issue #559 which mentions that some passwords require %!Password% to be used instead of %Password%. The link in that ticket is going nowhere, but I am assuming that this somehow escapes the password?
I did a cross-test by removing the ! from the password and that worked with the above string.

To make the string work with the ! password, I further changed the URL and it finally worked with:
`scp://%Username%:%!Password%@%Hostname%:%Port%/`

In summary, my suggestions are:
1) Change the URL string for WinSCP on https://mremoteng.readthedocs.io/en/latest/external_tools_cheat_sheet.html#id1 to `scp://%Username%:%!Password%@%Hostname%:%Port%/`
2) Document the existance of the hidden %!Password% placeholder

Thanks for working on this great tool!

**Comment by RolfWojtech** [comment]
I tested it further and it looks like different special characters cause different solutions to fail.

My original suggestion fails on  "password+".
The current "implementation" fails on "password!" and "password&"

The best solution I could come up with is combining both approaches (adding the exclamation mark to yours) and putting the password in quotes:
scp://%Username%@%Hostname%:%Port%/ /password="%!Password%"

This works with all the mentioned special characters.
I then generated the final boss password including all ASCII special characters:
"password!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~"

This still fails with all approaches. Some divide & conquer later I identified the double quotation marks (") as the culprit.

So the best solution does work for this password:
"password!#$%&'()*+,-./:;<=>?@[\]^_`{|}~"

Further research shows that WinSCP would need double quotation to be doubled ("") within the password to work:
https://winscp.net/eng/docs/scripting#quotes
https://winscp.net/forum/viewtopic.php?t=34372

Since this seems like special WinSCP behaviour, adding support for such passwords seems out of scope.

But adopting the fix to add the exclamation mark is required.

You should also document what !Password does, its not specified anywhere (I did not dive into the sources).

**Comment by RolfWojtech** [comment]
I should note that using the quotes before and after the %!Password% makes trailing quotes like "password"" work but not quotes within the password. I still added the quotes because its how the WinSCP member suggested to provide passwords in the second link and it works just as well.
